### DISCUSSION TITLE
[HASS Discovery] Notification CCs as  Binary sensors

### DISCUSSION THREAD

**Discussion by varet80** [zwave-js/zwave-js-ui#371] [Ideas]
Home assistant is providing two types of sensors. Binary and Multilevel/multistate sensors.

More Z-wave devices move toward notification CC! and stop using the Binary approach.

Some sensors like Motion, Flood or contact do have only two possible states Alert or Idle/off. 
I propose to migrate these sensors from sensor to binary type! and use the state as on/off state.

What we gain from is:
1. Icon can alert for true/false state (no complex setup).
2. easy to automate as you expect a true/false state.
3. Behaviour is similar to old sensors, and this make these grouped together

To make this rubust:
Any sensor having Idle state in their state list AND a maximum of two possible values, is the target of this setup.

Example of a sensor with only two states
```json
{
  "id": "88-113-0-Home Security-Motion sensor status",
  "nodeId": 88,
  "commandClass": 113,
  "commandClassName": "Notification",
  "endpoint": 0,
  "property": "Home Security",
  "propertyName": "Home Security",
  "propertyKey": "Motion sensor status",
  "propertyKeyName": "Motion sensor status",
  "type": "number",
  "readable": true,
  "writeable": false,
  "label": "Motion sensor status",
  "stateless": false,
  "ccSpecific": {
    "notificationType": 7
  },
  "min": 0,
  "max": 255,
  "list": true,
  "states": [
    {
      "text": "idle",
      "value": 0
    },
    {
      "text": "Motion detection",
      "value": 8
    }
  ],
  "value": 8,
  "lastUpdate": 1******,
  "nodeName": "Se***",
  "nodeLocation": "L***"
}
```

### Proposal
An automated way where if states are in total of two! and one state is Idle. Makes this possible. Any other case will fall back to normal sensor

**Comment by blhoward2** [comment]
The only thing I’d change is the maximum of two values. Water sensors actually have the potential for more I believe but anything other than idle signifies some sort of problem. I think the others are related to sump pumps IIRC. Other sensor types also have multiple potential trouble values. 

I think these sensors should have an alarm_state sensor that is binary no matter the status to indicate the alarm has triggered. This is less critical than an ordinary contact binary as most won’t have these. 

Note that this proposal is the same as many devices do setup now. My door sensors have both a binary_contact sensor and a sensor that says “Door is Opened” or “Door is closed.” These serve different purposes.

**Reply by varet80** [reply]
Binary sensors are only accepting two payloads, what you describe is multilevel sensor. And first priority is to cover most of the binary sensors.

The others should follow another approach.
Still having some idea for making these easier to get.
